hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/TestFileUtil.java

```diff
@@ -726,7 +727,10 @@ public void testUnZip() throws IOException {
     // check result:
     assertTrue(new File(tmp, "foo").exists());
     assertEquals(12, new File(tmp, "foo").length());
-    
+    assertTrue(new File(tmp, "foo").canExecute());
+    assertTrue(new File(tmp, "foo").canRead());
+    assertTrue(new File(tmp, "foo").canWrite());
```

Comment:
1. there's no tests to verify that other/group permissions are picked up. not sure if it is possible.
2. what happens if someone runs the test on windows?

Reply:
1.This is a good test scenario and I will add it.
2.If it is a Windows operating platform, it will skip the permission logic, because the Windows system itself does not have detailed executable permission, only read and write permission,the method of canExecute defaults to true on Windows systems.